src/neo/Network/P2P/TaskManager.cs

```diff
@@ -17,95 +17,72 @@ namespace Neo.Network.P2P
     public class TaskManager : UntypedActor
     {
         internal class Register { public VersionPayload Version; }
-        internal class Update { public uint LastBlockIndex; public bool RequestTasks; }
+        internal class Update { public uint LastBlockIndex; }
         internal class NewTasks { public InvPayload Payload; }
         public class RestartTasks { public InvPayload Payload; }
         private class Timer { }
 
         private static readonly TimeSpan TimerInterval = TimeSpan.FromSeconds(30);
         private static readonly TimeSpan TaskTimeout = TimeSpan.FromMinutes(1);
-        private static readonly UInt256 MemPoolTaskHash = UInt256.Parse("0x0000000000000000000000000000000000000000000000000000000000000001");
+        private static readonly UInt256 HeaderTaskHash = UInt256.Zero;
 
         private const int MaxConncurrentTasks = 3;
-        private const int MaxSyncTasksCount = 50;
-        private const int PingCoolingOffPeriod = 60_000; // in ms.
 
         private readonly NeoSystem system;
         /// <summary>
         /// A set of known hashes, of inventories or payloads, already received.
         /// </summary>
         private readonly HashSetCache<UInt256> knownHashes;
-        private readonly Dictionary<UInt256, int> globalTasks = new Dictionary<UInt256, int>();
-        private readonly Dictionary<uint, TaskSession> receivedBlockIndex = new Dictionary<uint, TaskSession>();
-        private readonly HashSet<uint> failedSyncTasks = new HashSet<uint>();
+        private readonly Dictionary<UInt256, int> globalInvTasks = new Dictionary<UInt256, int>();
+        private readonly Dictionary<uint, int> globalIndexTasks = new Dictionary<uint, int>();
         private readonly Dictionary<IActorRef, TaskSession> sessions = new Dictionary<IActorRef, TaskSession>();
         private readonly ICancelable timer = Context.System.Scheduler.ScheduleTellRepeatedlyCancelable(TimerInterval, TimerInterval, Context.Self, new Timer(), ActorRefs.NoSender);
-        private uint lastTaskIndex = 0;
+
+        private bool HasHeaderTask => globalInvTasks.ContainsKey(HeaderTaskHash);
 
         public TaskManager(NeoSystem system)
         {
             this.system = system;
             this.knownHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 5);
-            this.lastTaskIndex = NativeContract.Ledger.CurrentIndex(Blockchain.Singleton.View);
             Context.System.EventStream.Subscribe(Self, typeof(Blockchain.PersistCompleted));
             Context.System.EventStream.Subscribe(Self, typeof(Blockchain.RelayResult));
```

Comment:
@Qiao-Jin You need to check these events. It seams that this logic is removed by mistake.

Reply:
Please have a check

Reply:
@erikzhang